controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +458,277 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
 		).
 		Complete(r)
 }
+
+// ------------------------------synthetic-monitoring--------------------------
+// helper function to make Dynatrace api requests
+func (APIClient *APIClient) makeRequest(method, path string, renderedJSON string) (*http.Response, error) {
+	url := APIClient.baseURL + path
+	var reqBody io.Reader
+	if renderedJSON != "" {
+		reqBody = bytes.NewBufferString(renderedJSON)
+	}
+
+	req, err := http.NewRequest(method, url, reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Authorization", "Api-Token "+APIClient.apiToken)
+	req.Header.Set("Content-Type", "application/json")
+
+	return APIClient.httpClient.Do(req)
+}
+
+func (r *HostedControlPlaneReconciler) getSecret(ctx context.Context) (string, string, error) {
```

Comment:
I think this function name is also a bit too generic. Potentially we could rename this to something like `getDynatraceSecrets` or something?